src/components/Auth/Register.tsx

```diff
@@ -1,21 +1,48 @@
+import { createUserWithEmailAndPassword, updateProfile } from 'firebase/auth';
+import { Timestamp } from 'firebase/firestore';
+import { useState } from 'react';
 import { useForm } from 'react-hook-form';
 import { Link } from 'react-router-dom';
 import { LoginData } from '../../types';
+import { auth } from '../configuration/firebase';
 import AuthImages from '../helpers/AuthImages';
+import { UseSetDoc } from '../hooks/useManageDoc';
 
-interface RegisterProps {
-  registerError: string;
-  registerNewUser: (data: LoginData) => void;
-}
-
-function Register({ registerError, registerNewUser }: RegisterProps) {
+function Register() {
+  const [registerError, setRegisterError] = useState('');
   const {
     register,
     handleSubmit,
     formState: { errors },
   } = useForm<LoginData>();
 
-  const onSubmit = (data: LoginData) => registerNewUser(data);
+  const onSubmit = (data: LoginData) =>
+    createUserWithEmailAndPassword(auth, data.email, data.password)
+      .then(async (userCredential) => {
+        const { user } = userCredential;
+
+        await updateProfile(user, {
+          displayName: data.name,
+          photoURL: 'https://remaxgem.com/wp-content/themes/tolips/images/placehoder-user.jpg',
+        });
+
+        await UseSetDoc('users', [user.uid], {
+          uid: user.uid,
+          name: data.name,
+          email: user.email,
```

Comment:
{ ...user, createdAt: Timestamp.fromDate(new Date()), isOnline: true}
? Czy może obiekt usera zawiera zbyt dużo niepotrzebnych w tym miejscu kluczy? Jeśli tak, to może zdestrukturyzować go?

Reply:
ten user to zwrotka z Auth wiec tam jest zbyt duzo informacji których nie chcę zapisywac w bazie potrzebuję tylko uid i email z tego usera. Ale masz rację destrukturyzację mogę zrobic